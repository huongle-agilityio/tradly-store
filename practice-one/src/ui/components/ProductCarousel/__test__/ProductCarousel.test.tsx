import {
  render,
  fireEvent,
  screen,
  waitFor,
} from '@testing-library/react-native';

// Components
import { ProductCarousel } from '..';

describe('ProductCarousel', () => {
  const mockImages = ['image1.jpg', 'image2.jpg', 'image3.jpg'];

  it('Should renders correct number of images', () => {
    const { toJSON } = render(
      <ProductCarousel name="Test Product" images={mockImages} />,
    );

    expect(screen.getAllByTestId('product-carousel-image')).toHaveLength(
      mockImages.length,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should renders correct number of dots', () => {
    render(<ProductCarousel name="Test Product" images={mockImages} />);

    expect(screen.getAllByTestId('product-carousel-dot-item')).toHaveLength(
      mockImages.length,
    );
  });

  it('Should update container width on layout change', async () => {
    render(<ProductCarousel name="Test Product" images={mockImages} />);

    const container = screen.getByTestId('product-carousel-container');

    fireEvent(container, 'layout', { nativeEvent: { layout: { width: 300 } } });

    await waitFor(() => {
      const images = screen.getAllByTestId('product-carousel-image');
      images.forEach((image) => {
        expect(image.props.style.width).toBe(300);
      });
    });
  });

  it('Should triggers onScroll event when scrolling', () => {
    render(<ProductCarousel name="Test Product" images={mockImages} />);

    const scrollView = screen.getByTestId('product-carousel-scroll');

    fireEvent.scroll(scrollView, {
      nativeEvent: { contentOffset: { x: 200 } },
    });

    expect(scrollView.props.onScroll).toBeDefined();
  });
});
