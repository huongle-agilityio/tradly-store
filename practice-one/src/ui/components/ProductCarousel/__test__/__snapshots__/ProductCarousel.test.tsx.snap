// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCarousel Should renders correct number of images 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "backgroundColor": "rgba(8, 11, 18, 0.3)",
      "height": "100%",
      "position": "relative",
    }
  }
  testID="product-carousel-container"
>
  <RCTScrollView
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    testID="product-carousel-scroll"
  >
    <View>
      <ViewManagerAdapter_ExpoImage
        accessibilityLabel="Product image of Test Product - View 1"
        containerViewRef={"[React.ref]"}
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "height": 16,
              "uri": "blurhash:/%7CrF%3FhV%252WCj%5Bayj%5Ba%7Cj%5Baz_NaeWBj@ayfRayfQfQM%7BM%7Cazj%5Bazf6fQfQfQIpWXofj%5Bayj%5Bj%5BfQayWCoeoeaya%7Dj%5BayfQa%7BoLj%3Fj%5BWVj%5Bayayj%5BfQoff7azayj%5Bayj%5Bj%5Bayofayayayj%5BfQj%5Bayayj%5Bayfjj%5Bj%5Bayjuayj%5B",
              "width": 16,
            },
          ]
        }
        source={
          [
            {
              "uri": "image1.jpg",
            },
          ]
        }
        style={
          {
            "width": 0,
          }
        }
        testID="product-carousel-image"
        transition={
          {
            "duration": 1000,
          }
        }
        width={0}
      />
      <ViewManagerAdapter_ExpoImage
        accessibilityLabel="Product image of Test Product - View 2"
        containerViewRef={"[React.ref]"}
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "height": 16,
              "uri": "blurhash:/%7CrF%3FhV%252WCj%5Bayj%5Ba%7Cj%5Baz_NaeWBj@ayfRayfQfQM%7BM%7Cazj%5Bazf6fQfQfQIpWXofj%5Bayj%5Bj%5BfQayWCoeoeaya%7Dj%5BayfQa%7BoLj%3Fj%5BWVj%5Bayayj%5BfQoff7azayj%5Bayj%5Bj%5Bayofayayayj%5BfQj%5Bayayj%5Bayfjj%5Bj%5Bayjuayj%5B",
              "width": 16,
            },
          ]
        }
        source={
          [
            {
              "uri": "image2.jpg",
            },
          ]
        }
        style={
          {
            "width": 0,
          }
        }
        testID="product-carousel-image"
        transition={
          {
            "duration": 1000,
          }
        }
        width={0}
      />
      <ViewManagerAdapter_ExpoImage
        accessibilityLabel="Product image of Test Product - View 3"
        containerViewRef={"[React.ref]"}
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "height": 16,
              "uri": "blurhash:/%7CrF%3FhV%252WCj%5Bayj%5Ba%7Cj%5Baz_NaeWBj@ayfRayfQfQM%7BM%7Cazj%5Bazf6fQfQfQIpWXofj%5Bayj%5Bj%5BfQayWCoeoeaya%7Dj%5BayfQa%7BoLj%3Fj%5BWVj%5Bayayj%5BfQoff7azayj%5Bayj%5Bj%5Bayofayayayj%5BfQj%5Bayayj%5Bayfjj%5Bj%5Bayjuayj%5B",
              "width": 16,
            },
          ]
        }
        source={
          [
            {
              "uri": "image3.jpg",
            },
          ]
        }
        style={
          {
            "width": 0,
          }
        }
        testID="product-carousel-image"
        transition={
          {
            "duration": 1000,
          }
        }
        width={0}
      />
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "bottom": 15,
        "flexDirection": "row",
        "gap": 5,
        "position": "absolute",
        "zIndex": 999,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 9999,
          "height": 6,
          "opacity": 0.4,
          "width": 6,
        }
      }
      testID="product-carousel-dot-item"
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 9999,
          "height": 6,
          "opacity": 0.4,
          "width": 6,
        }
      }
      testID="product-carousel-dot-item"
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 9999,
          "height": 6,
          "opacity": 0.4,
          "width": 6,
        }
      }
      testID="product-carousel-dot-item"
    />
  </View>
</View>
`;
